### PR TITLE
feat(appstore): add more fields to request/response structs of app store server api

### PR DESCRIPTION
**Pull request by richzw**
feat(appstore): add more fields to request/response structs of app store server API per the latest app store server API document

### CODE REVIEW

**Comment by kaijietti** [comment]
I was planning to make these changes, thank you :+1:

**Review by takecy — COMMENTED** [review]
Thanks for your PR :)
Please check the comments 🙏

File: appstore/api/store.go [hunk 1]
```diff
@@ -301,17 +301,25 @@ func (a *StoreClient) ExtendSubscriptionRenewalDateForAll(ctx context.Context, b
 }
 
 // GetSubscriptionRenewalDataStatus https://developer.apple.com/documentation/appstoreserverapi/get_status_of_subscription_renewal_date_extensions
-func (a *StoreClient) GetSubscriptionRenewalDataStatus(ctx context.Context, productId, requestIdentifier string) (statusCode int, err error) {
+func (a *StoreClient) GetSubscriptionRenewalDataStatus(ctx context.Context, productId, requestIdentifier string) (rsp *MassExtendRenewalDateStatusResponse, err error) {
```
[the remark this answers is not in the document]
Comment (richzw) [reply appstore/api/store.go:304]: Thank you for your comment, I will fix it

[the remark this answers is not in the document]
Comment (richzw) [reply appstore/api/store.go]: Just fix it, Could you please review it again?

**Review by takecy — APPROVED** [review]
Thanks for the response.
LGTM 👍

**Pull request merged**